base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -730,16 +731,45 @@ fn add_block<T: BlockchainBackend>(
     handle_possible_reorg(metadata, db, block_validator, block)
 }
 
-fn store_new_block<T: BlockchainBackend>(db: &mut RwLockWriteGuard<T>, block: Block) -> Result<(), ChainStorageError> {
+fn store_new_block<T: BlockchainBackend>(
+    metadata: &mut RwLockWriteGuard<ChainMetadata>,
+    db: &mut RwLockWriteGuard<T>,
+    block: Block,
+) -> Result<(), ChainStorageError>
+{
     let (header, inputs, outputs, kernels) = block.dissolve();
+    let height = header.height;
+    let best_block = header.hash();
+    let accumulated_difficulty =
+        ProofOfWork::new_from_difficulty(&header.pow, ProofOfWork::achieved_difficulty(&header))
+            .total_accumulated_difficulty();
     // Build all the DB queries needed to add the block and the add it atomically
     let mut txn = DbTransaction::new();
+    // Update db metadata
+    txn.insert(DbKeyValuePair::Metadata(
+        MetadataKey::ChainHeight,
+        MetadataValue::ChainHeight(Some(height)),
+    ));
+    txn.insert(DbKeyValuePair::Metadata(
+        MetadataKey::BestBlock,
+        MetadataValue::BestBlock(Some(best_block.clone())),
+    ));
+    txn.insert(DbKeyValuePair::Metadata(
+        MetadataKey::AccumulatedWork,
+        MetadataValue::AccumulatedWork(Some(accumulated_difficulty)),
```

Comment:
Will this be reset after if reorg fails?

Reply:
Yes, then we will rewind the chain and add the original block back.